### ISSUE TITLE
Flaky: TestDQ_always_pop_earliest_deadline_multi

### ISSUE THREAD

**Issue by mikedanese** [kubernetes/kubernetes#11821]
I've seen this 3 times this morning.
```
--- FAIL: TestDQ_always_pop_earliest_deadline_multi (45.44s)
delay_test.go:296: Expected delay-prority of 1s got instead got 2s
FAIL
coverage: 64.9% of statements
FAIL	github.com/GoogleCloudPlatform/kubernetes/contrib/mesos/pkg/queue	56.855s
```

@jdef @karlkfi @sttts

**Comment by karlkfi** [comment]
yeah, that test is pretty terrible...

**Comment by karlkfi** [comment]
It's not really clear why that test would be taking 56 seconds. The only thing I can think of is that it yields several times and must be getting squeezed out of execution time. But if that's the case, the only way to make the test more reliable (without rewriting the DelayQueue) is to make the test take longer. And if that one test can get squeezed from 3seconds up to almost a minute, I'd have to make that one test take at least a minute to make it reliably unflakey.

**Comment by lavalamp** [comment]
Also see #11857. Can we disable these until this is resolved?

@karlkfi we've found that timings on shippable (or travis) are drastically different from local timings, so it's really not feasible to have timing-based tests. Waiting on a channel or other synchronization options are preferred.

**Comment by karlkfi** [comment]
If it's enough of an irritant, feel free to add a t.Skip("msg") to these two tests.
We'll leave these two issues open until we fix them.

**Comment by wojtek-t** [comment]
@jdef @karlkfi - what's the status of it? Is it fixed? If not, is someone looking into it?

**Comment by sttts** [comment]
@wojtek-t is it an issue again? I think we increased the timeouts, but it's not actually fixed AFAIK. Will bring that up again.

**Comment by sttts** [comment]
@wojtek-t do you have a link to a failed unit test run?

**Comment by wojtek-t** [comment]
No - I didn't see that for some time.
If you think it's fixed - I think we can close it, and open a new issue if we observe it again.

@sttts WDYT?

**Comment by sttts** [comment]
@wojtek-t while the tests are still dependent on a real time source (which is not really good), they were mainly a problem because Shippable used to take 56 sec for something trivial that was expected within 1-2 seconds. 

We have another ticket to make the code more testable: https://waffle.io/mesosphere/kubernetes-mesos. 

As a flake issue let's close this one.

**Comment by sttts** [comment]
Followed up by https://github.com/mesosphere/kubernetes-mesos/issues/432

**Comment by karlkfi** [comment]
The problem is more one of untestable design than bad tests. This is just a symptom of the delay queue needing rewriting
